flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,446 @@
+            FloodClient.authenticate(task, config)
+
+        entries = []
+        torrents = FloodClient.list_torrents(task, config)
+
+        for hash, torrent in torrents.items():
+            entry = self.generate_entry(config, torrent)
+
+            if entry:
+                entries.append(entry)
+
+        return entries
+
+
+class OutputFlood:
+    """Add, remove, delete, start, or stop torrents in Flood.
+
+    Example:
+        flood:
+            url: http://localhost:3000  # Required. Url for Flexget to connect to Flood.
+            username: flexget           # Required. Username for authentication with Flood.
+            password: password          # Required. Password for authentication with Flood.
+            action: add                 # Required. The action to perform. Can be 'add', 'remove', 'delete', 'start', or 'stop'.
+            path: /downloads            # If the action is set to 'add', the path is relative to the Flood download directory.
+            tags: [ 'Flexget' ]         # If the action is set to 'add', the tags to add to the torrent.
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'action': {'type': 'string', 'enum': ['add', 'remove', 'delete', 'start', 'stop']},
+            'path': {'type': 'string'},
+            'tags': {'type': 'array', 'items': {'type': 'string'}},
+        },
+        'additionalProperties': False,
+        'required': ['url', 'username', 'password', 'action'],
+    }
+
+    def _add_entry_file(
+        self, task: Task, config: dict, entry: Entry, destination: str, tags: list, start: bool
+    ) -> None:
+        """
+        Process an entry to add to Flood by file.
+        """
+
+        if not os.path.exists(entry['file']):
+            return entry.fail('Temp file not found.')
+
+        with open(entry['file'], 'rb') as fs:
+            encoded_file = base64.b64encode(fs.read()).decode('utf-8')
+
+            try:
+                FloodClient.add_torrent_files(
+                    task, config, [encoded_file], destination, tags, start
+                )
+            except PluginError as e:
+                return entry.fail(e)
+
+    def _add_entry_url(
+        self, task: Task, config: dict, entry: Entry, destination: str, tags: list, start: bool
+    ) -> None:
+        """
+        Process an entry to add to Flood by URL.
+        """
+
+        try:
+            FloodClient.add_torrent_urls(task, config, [entry['url']], destination, tags, start)
+        except PluginError as e:
+            return entry.fail(e)
+
+    def _process_accepted_entry(self, task: Task, config: dict, entry: Entry) -> None:
+        if config['action'] == 'add':
+            self.add_entry(task, config, entry)
+        elif 'flood_hash' in entry:
+            if config['action'] == 'remove':
```

Comment:
A pity you can't use `match` here, but it's only available in Python 3.10 and newer

Reply:
Unless you've got any other idea's for how I can handle different actions, if-else is probably here to stay.. haha